### PR TITLE
feat: handle active/expired status

### PR DESCRIPTION
**Pull request by jgellin-sf**
### What does this PR do?

This PR adds `ExperimentStatus` (`Active` or `Expired`) to the return type of `getExperiments`. `isExpired` and `randomAssignment` are refactored to a utilities module.

### What issues does this PR fix or reference?

@W-16360994@

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -12,14 +12,23 @@ export enum ExperimentType {
...
 }
 
+export interface ExperimentWithStatus extends Experiment {
+  status: ExperimentStatus;
+}
+
 export interface IExperimentService {
   registerExperiments(context: vscode.ExtensionContext, experiments: Experiment[]): void;
-  getExperiments(): Experiment[];
+  getExperiments(): ExperimentWithStatus[];
```
Comment (gbockus-sf) [inline src/api.ts:33]: Do we want consumers to has a type called ExperimentWithStatus?  
I'd change Experiment to ExperimentDefinition and make ExperimentWithStatus -> Experiment
thoughts?

File: test/unit/experimentService.test.ts [hunk 2]
```diff
@@ -22,8 +22,19 @@ describe('ExperimentService', () => {
...
       }
     ];
+    const expected = [
+      {
+        ...experiments[0],
+        status: 'active'
+      },
+      {
+        ...experiments[1],
+        status: 'active'
```
Comment (gbockus-sf) [inline test/unit/experimentService.test.ts:32]: nit: use the enum

**Review by gbockus-sf — DISMISSED** [review]
One consideration on type names...otherwise good to go.

File: src/api.ts (hunk 1 above)
Comment (jgellin-sf) [reply src/api.ts]: Yes. Thanks. I struggled with the naming. Fixed in https://github.com/forcedotcom/salesforcedx-vscode-experiments/pull/15/commits/100150d51f4d5083486898452942be0ad1e75abe.

File: test/unit/experimentService.test.ts (hunk 2 above)
Comment (jgellin-sf) [reply test/unit/experimentService.test.ts]: Fixed in https://github.com/forcedotcom/salesforcedx-vscode-experiments/pull/15/commits/a295ca1c60ed108e7eac78b1aa101ccd0f3cea41.

**Review by gbockus-sf — APPROVED** [review]
:shipit:

**Pull request merged**